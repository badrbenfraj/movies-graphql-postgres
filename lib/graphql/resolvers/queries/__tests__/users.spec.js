const users = require('../users')
const { db, } = require('../../../../adapters/db')

jest.mock('../../../../adapters/db', () => ({
  db: {
    query: jest.fn().mockReturnValue('dbData'),
  },
}))

describe('queries/users', () => {
  beforeEach(() => {
    db.query.mockClear()
  })

  it('should get users', () => {
    users(null, {})

    expect(db.query.mock.calls[0][0]).toMatchSnapshot()
  })

  it('should get a specific user', () => {
    users(null, { id: '1', })

    expect(db.query.mock.calls[0][0]).toMatchSnapshot()
  })

  it('should return database data', async () => {
    const response = await users(null, { id: '1', })

    expect(response).toEqual('dbData')
  })

  it('should handle errors', async () => {
    db.query.mockImplementation(() => Promise.reject('nope'))

    try {
      await users(null, { id: '1', })
    } catch (e) {
      expect(e.message).toMatch(/nope/)
    }
  })
})
