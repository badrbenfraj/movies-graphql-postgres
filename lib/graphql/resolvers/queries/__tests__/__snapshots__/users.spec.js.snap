// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries/users should get a specific user 1`] = `
"
    SELECT name, email, id FROM users
    WHERE id = 1
  "
`;

exports[`queries/users should get users 1`] = `
"
    SELECT name, email, id FROM users
    
  "
`;
